### ISSUE TITLE
When steal is bundled with an app, the CSS is loaded from the wrong path

### ISSUE THREAD

**Issue by dispatchrabbi** [stealjs/steal-tools#92]
With this HTML in /dist/bundles/index.html:

``` html
<!doctype html>
<html>
<head>
  <title>❄︎</title>
</head>
<body>
  <script src="./index.js"></script>
</html>
```

When I load /dist/bundles/index.html after building my app with `bundleSteal: true` in the Grunt task `buildOptions` config, this is what I see:
![screen shot 2014-10-02 at 4 37 13 pm](https://cloud.githubusercontent.com/assets/2257804/4500199/52f973be-4a8d-11e4-8531-234f5e5aece2.png)

It looks like the bundled version of steal isn't expecting to already be in /dist/bundles (where index.js is), and so it adds "/dist/bundles" to the beginning of the CSS path, causing a request to /dist/bundles/dist/bundles/index.css.

**Comment by whitecolor** [comment]
It is going to be fixed in upcomming release?

**Comment by justinbmeyer** [comment]
@whitecolor It is configurable.  This is what bundlesPaths does.

Is there something to be fixed? Can you create an example (or better test) of it not working?

**Comment by justinbmeyer** [comment]
@whitecolor It would be helpful if you could clearly explain the problem.  

Perhaps, the problem is that bundlesPath should not be configured from stealconfig.js.  It really must be configured from the attributes or a `steal` object in the page.

Some code examples or test would be good.  If you can get this today, I can put it in the 0.6.0 release.  Thanks!

cc @alexisabril @joe-crick (there might be some docs changes)

**Comment by whitecolor** [comment]
I'm kinda confused about what is wrong with this paths, but something definitly wrong.

One of the problems:
in my build config (for grunt task) i've got:

```javascript
config.stealBuild.mobile = {
        options: {
            system: {
                config: dirname + "/stealconfig.mobile.js",
                main: "mobile/mobile",
                bundlesPath: 'mobile/dist/www/assets'                
            },
            buildOptions: {
                bundleSteal: true
            }

        }
    }
```

```bundlesPath``` here is just a path where assets should be put. Why this paths go in the built code?

```javascript
/*[system-bundles-config]*/
System.paths["bundles/*.css"] ="mobile\dist\www\assets/*css";
System.paths["bundles/*"] = "mobile\dist\www\assets/*.js";
```

I would prefere to have something like that: some relative paths to main bundle (if i'm correct, this is puzzle for me too, what I should have in result file)
```javascript
/*[system-bundles-config]*/
System.paths["bundles/*.css"] ="./*css";
System.paths["bundles/*"] = "./*.js";
```


can you explai what is wrong with my config? How should I change it to have correct result?

**Comment by justinbmeyer** [comment]
> I would prefere to have something like that: some relative paths to main bundle (if i'm correct, this is puzzle for me too, what I should have in result file)

I don't know what you mean here.  Can you rewrite it?

**Comment by whitecolor** [comment]
On the destination (server or phonegap fs) I wll have following structure:
```
/(root)
    /assets/
    mobile.js (main bundle js)
    mobile.css (main bundle css)

index.html
```

in index.html I load main bundle

```HTML
<script src="./assets/mobile.js"></script>
```

What configuration of bundlesPath should I have to load mobile.js and mobile.css loaded correctly?

**Comment by justinbmeyer** [comment]
I'm confused a bit.  It seems obvious to me that by setting bundlesPaths to `mobile/dist/www/assets`, it would result in:

```js
System.paths["bundles/*.css"] ="mobile\dist\www\assets/*css";
System.paths["bundles/*"] = "mobile\dist\www\assets/*.js";
```

This is what bundlesPaths does, it controls where the bundles are written out (while keeping baseURL the same).

I think that baseURL is kept the same is what is confusing you. I think you want baseURL to be moved to `moble/dist/www/assets`.

**Comment by justinbmeyer** [comment]
> What configuration of bundlesPath should I have to load mobile.js and mobile.css loaded correctly?

You would probably have to do something like:

```
<script src="./assets/mobile.js" base-url="../../"></script>
```

Like @matthewp  suggested.

**Comment by whitecolor** [comment]
So you propose me to leave
```javascript
System.paths["bundles/*.css"] ="mobile\dist\www\assets/*css";
```
in prodution code?

For me 
```javascript
  bundlesPath: 'mobile/dist/www/assets'    
```

in build config should be just place where I want built files to be. Why it is somehow related to where they are going to be on the server? Server doesn't know anything about ```mobile/dist/www/``` - so if such thing is happen to be in built code that run in production it is incorrect by design, don't you think so? How can I get rid of it in my configuration?

Having something like:
```javascript
<script src="./assets/mobile.js" base-url="../../"></script>
```
seem like a bug workaround or hack to me, no?

**Comment by justinbmeyer** [comment]
@matthewp I think what people really want here is the ability to "move" where `baseURL` would be located.

This might cause all sorts of problems if they are:
 - `System.import`ing modules not in a bundle (that haven't been copied over to the new `baseURL`) 
 - Loading anything relative to `baseURL` that hasn't been copied over (like an img).

`bundlesPath` is the path from the current baseURL to where the bundles should be located.  It does not change the `baseURL`.


Actually, we can probably make it set `base-url` correctly in the written-out production.  I'm surprised it's not already doing that.  

However, I could see that people will try to load the bundles with "impossible" urls.  Meaning if someone set `bundlesPath` to "mobile/dist/www/assets", and hosted assets under: _http://domain.com/assets_, a baseURL of "../../" would be impossible.  

So, despite the problems I listed above, we still might need an option that would move baseURL, and allow a `bundlesPath` of "./".  The user will have to copy any other assets over themselves.

**Comment by justinbmeyer** [comment]
> Having something like ... seem like a bug workaround or hack to me, no?

No, because of the problems I listed above.  Anything your CSS / LESS / code references related to baseURL will be broken unless you also copy those things to your `dist` folder.

I'm on gitter: https://gitter.im/bitovi/steal.  Lets talk about it more there.

**Comment by matthewp** [comment]
We have bundleAssets now when I think solves this need, can we close this?
